### PR TITLE
Source Google Ads: Improve unit and integration tests

### LINKED ISSUE #12650: Source Google Ads: improve unit and integration tests
Goal:
 - unit test coverage at least 90%
 - mock unit tests so they do not use real network connections
 - all integration tests pass
 
 Note: some of the SATs are disabled due to current constraints, so custom integration tests are implemented instead. The [issue](https://github.com/airbytehq/airbyte/issues/12665) describes the problem and the solution. After it is implemented, SATs can be enabled

### PR DESCRIPTION
**Pull request by davydov-d**
- Mock unit tests
- Reimplement two custom integration tests that replace commented out SATs
- Speed up Dockerfile

 Note: some of the SATs are disabled due to current constraints, so custom integration tests are implemented instead. The [issue](https://github.com/airbytehq/airbyte/issues/12665) describes the problem and the solution. After it is implemented, SATs can be enabled

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]

**Review by midavadim — APPROVED** [review]
Update can be merged without Airbyte review

File: airbyte-integrations/connectors/source-google-ads/integration_tests/test_incremental.py [hunk 1]
```diff
@@ -3,93 +3,88 @@
 #
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-google-ads/integration_tests/test_incremental.py:3]: done

**Pull request merged**